tests/TestEntry.cpp

```diff
@@ -158,3 +160,109 @@ void TestEntry::testResolveUrl()
 
     delete entry;
 }
+
+void TestEntry::testResolveUrlPlaceholders()
+{
+    Entry* entry = new Entry();
+    entry->setUrl("https://user:pw@keepassxc.org:80/path/example.php?q=e&s=t+2#fragment");
+
+    QString rmvscm("//user:pw@keepassxc.org:80/path/example.php?q=e&s=t+2#fragment"); // Entry URL without scheme name.
+    QString scm("https"); // Scheme name of the entry URL.
+    QString host("keepassxc.org"); // Host component of the entry URL.
+    QString port("80"); // Port number of the entry URL.
+    QString path("/path/example.php"); // Path component of the entry URL.
+    QString query("q=e&s=t+2"); // Query information of the entry URL.
+    QString userinfo("user:pw"); // User information of the entry URL.
+    QString username("user"); // User name of the entry URL.
+    QString password("pw"); // Password of the entry URL.
+    QString fragment("fragment"); // Password of the entry URL.
```

Comment:
The comment here is wrong, just a typo :)

Reply:
Done